PowerValidatedSolutions.psm1

```diff
@@ -50441,6 +50441,469 @@ Function Test-WMSubnetInput {
+        if ((-Not(Test-Path -Path $cfgFilePath)) -and (-Not (Test-Path -Path $orginalKeyFilePath))) {
+			Write-Output "1" | Set-Content $cfgFilePath
+			Write-Output "1" | Set-Content $orginalKeyFilePath
+			if ((Test-Path -Path $cfgFilePath) -and (Test-Path -Path $orginalKeyFilePath)) {
+				Remove-Item $cfgFilePath
+				Remove-Item $orginalKeyFilePath
+				break
+			}
+        } else {
+			if($createRandomCounter -gt 6) {
+				Write-Error "Unable to create temporary files in directory $certDirPath." -ErrorAction Stop
+				Exit
+			}
+        }
+    }
+
+    $keyFilePath = Join-Path -path $certDirPath -ChildPath "$commonName.key"
+    $csrFilePath = Join-Path -path $certDirPath -ChildPath "$commonName.csr"
+    if (Test-Path -Path $keyFilePath) {
+        Write-Error "$keyFilePath file already exists.  Please remove or rename the current file or use an alternate directory path." -ErrorAction Stop
+    }
+    if (Test-Path -Path $csrFilePath) {
+        Write-Error "$csrFilePath file already exists.  Please remove or rename the current file or use an alternate directory path." -ErrorAction Stop
+    }
+    if ($expireDays -lt 2) {
+        Write-Error "Certificate expiration day ($expireDays) is less than 1 day.  Please provide a value greater than 1 day."
+    }
+
+    # Create temporary cfg file
+    $cfgString = '[ req ]
+    days = ' +$expireDays+ '
+    default_md = sha512
+    default_bits = ' +$keySize+ '
+    distinguished_name = req_distinguished_name
+    prompt = no
+    req_extensions = v3_req
+    
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment, dataEncipherment, nonRepudiation
+    extendedKeyUsage = serverAuth, clientAuth
+    subjectAltName = @alternate_names
+
+    [ req_distinguished_name ]
+    commonName = ' +$commonName+ '
+    countryName = ' +$country+ '
+    stateOrProvinceName = ' +$state+ '
+    localityName = ' +$locality+ '
+    0.organizationName = ' +$organization+ '
+    organizationalUnitName = ' +$organizationUnit+ '
+    
+    [ alternate_names ]
+    '
+    # Parsing SAN string
+    $sanArrary = $subjectAlternativeNames -split "\s*,\s*"
+    $sanCount = 1
+
+    foreach($san in $sanArrary) {
+        $cfgString = $cfgString + "DNS." + $sanCount + " = " + $san + "`n"
+        $sanCount++
+    }
+
+    # Generate RSA private key and CSR file
+    $cfgString | Out-File $cfgFilePath
+    openssl req -new -sha256 -nodes -out $csrFilePath -keyout $orginalKeyFilePath -config $cfgFilePath >> $null 2>$null
+    openssl rsa -in $orginalKeyFilePath -out $keyFilePath -traditional >> $null 2>$null
+
+    # Clean up
+    Remove-Item $cfgFilePath
+    Remove-Item $orginalKeyFilePath
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsr
+
+Function Invoke-RequestSignedCertificate {
+    <#
+        .SYNOPSIS
+        Request Certificate Signing Request (CSR) to be signed by a Microsoft Certificate Authority.
```

Comment:
```suggestion
        Request Certificate Signing Request (CSR) to be signed by a certificate authority.
```

Reply:
addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014